### PR TITLE
add: `xonotic`

### PR DESCRIPTION
**Pull request by spaceguybob**

### CODE REVIEW

File: packages/xonotic/xonotic.pacscript [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+breaks=("xonotic-git")
+arch=('amd64')
+maintainer="Herisson <spaceguybox@outlook.com>"
+repology=("project: xonotic")
+
+package() {
+  sudo install -Dm755 "xonotic-linux64-sdl" "${pkgdir}/usr/bin/${pkgname}-sdl"
+  sudo install -Dm755 "xonotic-linux64-glx" "${pkgdir}/usr/bin/${pkgname}-glx"
+  sudo mkdir -p "${pkgdir}/usr/share/applications/"
+  sudo mkdir -p "~/.xonotic"
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:17]: Can we not install anywhere else? Also use `${homedir}` instead of `~`

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/xonotic/xonotic.pacscript (hunk 1 above)
Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript:17]: no we can not, else xonotic does not launch and cannot find the data directory. we could in theory redirect it to another path and add a flag to move it but that requires us to write a new .desktop file entirely

Comment (Elsie19) [reply packages/xonotic/xonotic.pacscript]: Let's do that, because application data should never be in the home directory.

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript]: okay, it'll be way more annoying but sure.

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript]: @Elsie19 https://xonotic.org/faq/#config it wants it there for the configs though too

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript]: we could force -basedir for the game data but the configs may not be movable

Comment (Elsie19) [reply packages/xonotic/xonotic.pacscript]: That's good. Configs are ok to be left in the homedir, it's just I'd really prefer the app data not to be.

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript]: then we need to make a fresh .desktop with the exec as ```xonotic -basedir /somedir````

Comment (Elsie19) [reply packages/xonotic/xonotic.pacscript]: https://github.com/pacstall/pacstall/wiki/Pacscript-101#other

Check here for how to add a desktop file.

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript]: thank you

File: packages/xonotic/xonotic.pacscript [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+repology=("project: xonotic")
+
+package() {
+  sudo install -Dm755 "xonotic-linux64-sdl" "${pkgdir}/usr/bin/${pkgname}-sdl"
+  sudo install -Dm755 "xonotic-linux64-glx" "${pkgdir}/usr/bin/${pkgname}-glx"
+  sudo mkdir -p "${pkgdir}/usr/share/applications/"
+  sudo mkdir -p "${homedir}/.xonotic"
+  sudo mkdir -p "${pkgdir}/.xonotic"
+  cp ~/.* ${pkgdir}/.xonotic/
+        echo '[Desktop Entry]
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:20]:
```suggestion
        echo "[Desktop Entry]
```

File: packages/xonotic/xonotic.pacscript [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Name=Xonotic
+Comment=Multiplayer, deathmatch oriented first person shooter
+Comment[de]=Deathmatch- und Mehrspieler-orientierter Ego-Shooter
+Comment[fr]=Jeu de tir à la première personne multijoueur
+Icon=xonotic
+Exec=xonotic-sdl -basedir ${pkgdir}/.xonotic
+Terminal=false
+PrefersNonDefaultGPU=true
+StartupNotify=false
+Categories=Game;ActionGame;' | sudo tee "${pkgdir}/usr/share/applications/xonotic.desktop" > /dev/null
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:32]:
```suggestion
Categories=Game;ActionGame;" | sudo tee "${pkgdir}/usr/share/applications/xonotic.desktop" > /dev/null
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/xonotic/xonotic.pacscript [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+maintainer="Herisson <spaceguybox@outlook.com>"
+repology=("project: xonotic")
+
+package() {
+  sudo install -Dm755 "xonotic-linux64-sdl" "${pkgdir}/usr/bin/${pkgname}-sdl"
+  sudo install -Dm755 "xonotic-linux64-glx" "${pkgdir}/usr/bin/${pkgname}-glx"
+  sudo mkdir -p "${pkgdir}/usr/share/applications/"
+  sudo mkdir -p "${homedir}/.xonotic"
+  sudo mkdir -p "${pkgdir}/.xonotic"
+  cp ~/.* ${pkgdir}/.xonotic/
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:19]: What is this line doing?

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/xonotic/xonotic.pacscript (hunk 4 above)
Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript:19]: that one could probably be deleted

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript:19]: you mean like 16 right?

Comment (Elsie19) [reply packages/xonotic/xonotic.pacscript:19]: No, I meant why are you copying all the hidden folders in the home directory anywhere

Comment (spaceguybob) [reply packages/xonotic/xonotic.pacscript:19]: oh god i didn't see that :sob: it's meant to be ./* as in the xonotic data sorry

File: packages/xonotic/xonotic.pacscript [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+maintainer="Herisson <spaceguybox@outlook.com>"
+repology=("project: xonotic")
+
+package() {
+  sudo install -Dm755 "xonotic-linux64-sdl" "${pkgdir}/usr/bin/${pkgname}-sdl"
+  sudo install -Dm755 "xonotic-linux64-glx" "${pkgdir}/usr/bin/${pkgname}-glx"
+  sudo mkdir -p "${pkgdir}/usr/share/applications/"
+  sudo mkdir -p "${homedir}/.xonotic"
+  sudo mkdir -p "${pkgdir}/.xonotic"
+  cp /.* ${pkgdir}/.xonotic/
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:19]:
```suggestion
  cp ./.* ${pkgdir}/.xonotic/
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/xonotic/xonotic.pacscript [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+arch=('amd64')
+maintainer="Herisson <spaceguybox@outlook.com>"
+repology=("project: xonotic")
+
+package() {
+  sudo install -Dm755 "xonotic-linux64-sdl" "${pkgdir}/usr/bin/${pkgname}-sdl"
+  sudo install -Dm755 "xonotic-linux64-glx" "${pkgdir}/usr/bin/${pkgname}-glx"
+  sudo mkdir -p "${pkgdir}/usr/share/applications/"
+  sudo mkdir -p "${pkgdir}/usr/share/xonotic"
+  sudo cp ./* ${pkgdir}/usr/share/xonotic/ -r
```
Comment (Elsie19) [inline packages/xonotic/xonotic.pacscript:18]:
```suggestion
  sudo cp ./* "${pkgdir}/usr/share/xonotic/" -r
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

**Review by oklopfer — APPROVED** [review]

**Pull request merged**